src/pycell/impl_.rs

```diff
@@ -284,6 +284,7 @@ mod tests {
             )
             .unwrap();
 
+            #[allow(deprecated)]
             let mmm_cell: &PyCell<MutableChildOfMutableChildOfMutableBase> = mmm.as_ref(py);
```

Comment:
I think in this file we should be able to change to use `Bound::borrow_mut` and `Bound::borrow` trivially.